File: src/1441.py
class Solution:
    def buildArray(self, target: list[int], n: int) -> list[str]:
        ops = []
        cur = 1
        for i in range(len(target)):
            while (cur < target[i]):
                ops.append("Push")
                ops.append("Pop")
                cur += 1
            ops.append("Push")
            cur += 1
        return ops

File: src/1441.cpp
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> ans;
        int cur = 1;
        for(int i = 0; i < target.size(); ++i) {
            while(cur < target[i]) {
                ans.push_back("Push");
                ans.push_back("Pop");
                cur++;
            }
            ans.push_back("Push");
            cur++;
        }
        return ans;
    }
};
